src/Validation/gen/ValidationsGenerator.cs

```diff
@@ -16,21 +19,38 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             predicate: FindAddValidation,
             transform: TransformAddValidation
         );
-        // Extract types that have been marked with [ValidatableType].
-        var validatableTypesWithAttribute = context.SyntaxProvider.ForAttributeWithMetadataName(
+
+        // Extract types that have been marked with framework [ValidatableType].
+        var frameworkValidatableTypes = context.SyntaxProvider.ForAttributeWithMetadataName(
             "Microsoft.Extensions.Validation.ValidatableTypeAttribute",
             predicate: ShouldTransformSymbolWithAttribute,
             transform: TransformValidatableTypeWithAttribute
         );
+
+        // Extract types that have been marked with generated [ValidatableType].
+        var generatedValidatableTypes = context.SyntaxProvider.ForAttributeWithMetadataName(
+            "Microsoft.Extensions.Validation.Embedded.ValidatableTypeAttribute",
+            predicate: ShouldTransformSymbolWithAttribute,
+            transform: TransformValidatableTypeWithAttribute
+        );
+
+        // Combine both sources of validatable types
+        var validatableTypesWithAttribute = frameworkValidatableTypes
```

Comment:
I think you can use the existing `Concat` helper method from [here](https://github.com/dotnet/aspnetcore/blob/828955caf1442ea13383fe0d2a75d388b8c54215/src/Validation/gen/Extensions/IncrementalValuesProviderExtensions.cs#L40).